torch/utils/data/dataframes_pipes.ipynb

```diff
@@ -0,0 +1,302 @@
+     ]
+    }
+   ],
+   "source": [
+    "dp = get_dataframes_pipe(batch_size = 3)\n",
+    "dp['y'] = dp.i * 100 + dp.j - 2.7\n",
+    "dp = dp.shuffle()\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "You can continue mixing DF and DP operations"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "    i   j          y\n0 -11 -17  5803000.0\n1 -10 -16  6813000.0\n2  -9 -15  7823000.0\n    i   j          y\n0 -14 -17  2803000.0\n1 -13 -16  3813000.0\n2 -12 -15  4823000.0\n   i   j           y\n0  1 -17  17803000.0\n1  2 -16  18813000.0\n   i   j           y\n0 -8 -17   8803000.0\n1 -7 -16   9813000.0\n2 -6 -15  10823000.0\n   i   j           y\n0 -2 -17  14803000.0\n1 -1 -16  15813000.0\n2  0 -15  16823000.0\n   i   j           y\n0 -5 -17  11803000.0\n1 -4 -16  12813000.0\n2 -3 -15  13823000.0\n    i   j          y\n0 -17 -17  -197000.0\n1 -16 -16   813000.0\n2 -15 -15  1823000.0\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = get_dataframes_pipe(batch_size = 3)\n",
+    "dp['y'] = dp.i * 100 + dp.j - 2.7\n",
+    "dp = dp.shuffle()\n",
+    "dp = dp - 17\n",
+    "dp['y'] = dp.y * 10000\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "#### Open questions:\n",
+    " - Does this `bath` API looks intuitive?"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "batch 0\n    i   j          y\n0 -11 -17  5803000.0\n1 -10 -16  6813000.0\n2  -9 -15  7823000.0\n    i   j          y\n0 -14 -17  2803000.0\n1 -13 -16  3813000.0\n2 -12 -15  4823000.0\nbatch 1\n   i   j           y\n0 -2 -17  14803000.0\n1 -1 -16  15813000.0\n2  0 -15  16823000.0\n   i   j           y\n0 -8 -17   8803000.0\n1 -7 -16   9813000.0\n2 -6 -15  10823000.0\nbatch 2\n    i   j          y\n0 -17 -17  -197000.0\n1 -16 -16   813000.0\n2 -15 -15  1823000.0\n   i   j           y\n0 -5 -17  11803000.0\n1 -4 -16  12813000.0\n2 -3 -15  13823000.0\nbatch 3\n   i   j           y\n0  1 -17  17803000.0\n1  2 -16  18813000.0\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = get_dataframes_pipe(batch_size = 3)\n",
+    "dp['y'] = dp.i * 100 + dp.j - 2.7\n",
+    "dp = dp.shuffle()\n",
+    "dp = dp - 17\n",
+    "dp['y'] = dp.y * 10000\n",
+    "dp = dp.batch(2)\n",
+    "for i,v in enumerate(dp):\n",
+    "    print('batch',i)\n",
+    "    for x in v:\n",
+    "        print(x)"
+   ]
```

Comment:
For `batch` over DFPipe, we can take a look here https://github.com/facebookexternal/torchdata/blob/main/torcharrow/torcharrow-batching.ipynb
to yield `df[0: batch_size], df[batch_size:batch_size*2], ...`

We can use `batch_size` as an internal state for DFPipe without actually split them into multiple DFs. So, we can still run ops over the whole DF. (It's kind like push batching logic to the end of the DF pipeline)

In this case, `batch` op over DFPipe won't change DFPipe object into regular DataPipe.